### ISSUE TITLE
A better module name ?

### ISSUE THREAD

**Issue by tst2005** [Stepets/utf8.lua#12]
Hello,

You probably know that since lua 5.3 a module named `utf8` is already used ( https://www.lua.org/manual/5.3/manual.html#6.5 )
I like your project. I think it should be more widly use to support string operation with utf-8 data.

- Do you planned to make a rockspec and publish your module on luarocks ?

- Can you consider to find a better repository name ?

I'm working on :
1) make a simple rockspec
2) make an all-in-one module file (my util to do that usually use the rockspec file)
In my current tries, I use `stepets_utf8` as module name. (`require "stepets_utf8"`)

Another name could be `utf8string` because you mainly implement all lua `string.*` features with utf8 support.
This `utf8string` package name is [not yet used on luarocks](https://luarocks.org/search?q=utf8string)

What is your opinion ?

**Comment by Stepets** [comment]
Hello, I don't have time to make a rockspec and luarocks publishing but these definitely are nice to have things.

I believe that 'utf8.lua' is the best possible name for this github repository. Because it's minimal and self-explanatory.
As I mentioned in the readme there is [the way](https://github.com/Stepets/utf8.lua#installation) to 'require' a module from 'utf8' subdirectory. But a user isn't forced to place this project in a directory named like this. As in your case it can be placed under any possible name.

If you are concerned about luarocks project name, then, yeah, utf8string is pretty nice name. Feel free to update [installation](https://github.com/Stepets/utf8.lua#installation) instructions.

**Comment by tst2005** [comment]
Even I did'nt answer for a while. Yesterday I successfully made an all-in-one file module (named `utf8string.lua`). Currently without using the rockspec made because the rockspec use string key in the module list, and lua by design didn't reming the key order then I use a custom way to list the module and their order. It was a neigthmare to compare results when the content are always placed on different order.
Now I will work to use the rockspec ;-)
You can close this issue, but I'm still working on it ;-)

**Comment by Stepets** [comment]
Whoa! Nice to hear it! Have a good luck with rockspec!

**Comment by tst2005** [comment]
If you are curious...
I put the [utf8string.lua](https://github.com/tst2005contrib/utf8string/blob/allinone-test/utf8string.lua) all-in-one module.
It pass [the test](https://github.com/tst2005contrib/utf8string/tree/allinone-test) (lua 5.1/5.2/5.3/jit) !

**Comment by Stepets** [comment]
Good job! Did you consider publishing script that packs all files in one? It might be useful as there already was [question](https://github.com/Stepets/utf8.lua/issues/10) about it.

**Comment by tst2005** [comment]
Yes sure!
A long time ago, I already made an util like that.
I had add more an more features and finally It become a nigthmare to maintained.
My own dev version was totally broken and too complexe to debug easily.

I decided to write a new one from scratch.
The main idea was : I will use templates.
Then I write my own template engine!

On this 2nd util that I named `allinone`...
I experimented plenty of different ways to pack... and make my template engine more an more powerfull.

I will release them soon.

The allinone util is not too hard because lot of stuff are done by the template engine.
My template engine is still in active development.
I stopped to add new feature in my template engine and use the current code to make allinone running successfully.

Yesterday, I already made another step : I made an all-in-one version of the allinone util itself !
Even it is run, yearh ! cool!  I think it is not the right way to do.
My wider approach is to make all-in-one version of each module. After that make a bundle taht will contains the allinone and all the dependencies.

- [allinone](https://github.com/lua-sword/allinone)
- [mini-tmpl](https://github.com/lua-sword/mini-tmpl) (oh, last commit "4 years ago". of course the recent code was not released yet)

Thanks for your answer, it make me feel less lonely on my projects!
